internal/cache.go

```diff
@@ -0,0 +1,10 @@
+package internal
+
+var cacheFolder = "weather-applet"
+
+func GetCacheDir(xdg_data_dirs string) string {
```

Comment:
It should be `XDG_DATA_HOME` instead of `XDG_DATA_DIRS`

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html